python/pyspark/serializers.py

```diff
@@ -64,7 +64,7 @@
     from itertools import izip as zip
 else:
     import pickle
-    protocol = 3
+    protocol = min(pickle.HIGHEST_PROTOCOL, 4)
```

Comment:
Should we only change the protocol for Broadcast? Otherwise we should also make sure that the Pyrolite can support protocol 4, which is used by Spark SQL.

Reply:
It sounds like this part of the change would be ok then since Pyrolite can read v4 just fine.